pkg/build/interfaces.go

```diff
@@ -49,6 +49,12 @@ type Ignorer interface {
 	Ignore(*api.Config) error
 }
 
+// Runner allows the user to actually run the produced imgae as part of the s2i
+// cmd line invocation, to see if it operates as he would expect
+type Runner interface {
+	Run(*api.Config) error
+}
```

Comment:
Yes, but these interfaces specifically are for builds only (pkg/**build**/interfaces.go). And as long as providing an interface makes sense here you have only one implementation of that interface and you're not using the interface directly anywhere, so generally speaking you don't leverage the power of that interface.